### PR TITLE
Fixed a bug where erroneous costs could be assigned to pre-cast spells

### PR DESCRIPTION
**Pull request by kfinch**
Caused by classResource field being copied with cost included into pre-cast spells - see bug in action here with a potion being listed as costing Astral Power: https://wowanalyzer.com/report/vra49mt8BjdPwDGZ/23-Mythic+Lady+Inerva+Darkvein+-+Kill+(5:33)/Coolbeanss/standard/astral-power-usage

### CODE REVIEW

**Comment by MartijnHols** [comment]
Hey are the precommit hooks not running automatically for you? Is this something I need to investigate?

File: src/parser/shared/normalizers/PrePullCooldowns.js [hunk 1]
```diff
@@ -140,8 +140,14 @@ class PrePullCooldowns extends EventsNormalizer {
          * information more accurately.
          */
         if (precastClassResources === null && event.classResources) {
-          debug && console.debug('Setting prepull class resources to:', event.classResources);
+          debug && console.debug('Setting prepull class resources (but stripping costs) to:', event.classResources);
           precastClassResources = event.classResources;
+          // make a new class resources with costs stripped - have to deep copy to avoid messing with the original cast event
+          precastClassResources = precastClassResources.map(classResource => {
+            const noCostClassResource = Object.assign({}, classResource);
```
Comment (MartijnHols) [inline src/parser/shared/normalizers/PrePullCooldowns.js]: In the modern JS version we use, you can write this like so:

```
const noCostClassResource ={
  ...classResource,
  cost: undefined,
};
```

With that this entire section could become:

```
precastClassResources = precastClassResources.map((classResource) =>({
  ...classResource,
  cost: undefined,
}))
```

Comment (kfinch) [reply src/parser/shared/normalizers/PrePullCooldowns.js]: Right I forgot you could do that, will change it

**Comment by kfinch** [comment]
Looks like the integration test diff is because this bug was manifesting in a very minor way in the MM hunter log, so I'll just update the test

**Pull request merged**